sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Metrics/ExecutableLinesWalkerTest.cs

```diff
@@ -328,5 +329,114 @@ static void Main(string[] args)
                 }",
                 4);
         }
+
+        [TestMethod]
+        public void Test_18_ExcludeFromTestCoverage()
+        {
+            AssertLinesOfCode(
+              @"using System.Diagnostics.CodeAnalysis;
+                namespace project_1
+                {
+                    public class ComplicatedCode
+                    {
+                        [ExcludeFromCodeCoverage]
+                        public string ComplexFoo()
+                        {
+                            string text = null;
+                            return text.ToLower();
+                        }
+
+                        [SomeAttribute]
+                        public string ComplexFoo2()
+                        {
+                            string text = null;
+                            return text.ToLower(); // +1
+                        }
+                    }
+                }",
+                17);
+        }
+
+        [TestMethod]
+        public void Test_19_ExcludeFromTestCoverage_Variants()
+        {
+            var attributeVariants = new List<string>
+            {
+                "ExcludeFromCodeCoverage",
+                "ExcludeFromCodeCoverage()",
+                "ExcludeFromCodeCoverageAttribute",
+                "ExcludeFromCodeCoverageAttribute()",
+                "System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute()"

```

Comment:
Add a test with something like `Foo.Bar.CanExcludeFromCodeCoverageAttribute` and you will see this is going to provide a wrong result.